packages/astro/src/runtime/server/render/util.ts

```diff
@@ -207,6 +207,8 @@ export function renderToBufferDestination(bufferRenderFunction: RenderFunction):
 
 export const isNode =
 	typeof process !== 'undefined' && Object.prototype.toString.call(process) === '[object process]';
+// @ts-expect-error: Deno is not part of the types.
+export const isDeno = typeof Deno !== 'undefined';
```

Comment:
Not something that cannot be bypassed unfortunately.

Reply:
There is a specific feature test you can do for the `new Response(asyncIterable)` behaviour - I'll post a feature test for this shortly.

Reply:
You can use this feature test to ensure this code will not break for runtimes that do not implement this behaviour. This will also shield you if Node will unship this behaviour in the future.

```js
const supportsResponseBodyAsyncIterator = (() => {
  let supported = false;
  new Response({
    get [Symbol.asyncIterator]() {
      supported = true;
      return undefined;
    },
  });
  return supported;
})();
```

```
❯❯❯ deno run test.js
false
❯❯❯ node test.js
true
❯❯❯ bun test.js
true

## chrome
false
```